### PR TITLE
Issue #5430: provide display settings for book navigation

### LINKED ISSUE #5430: Added settings for which parts of the book navigation to display.
## Description of the need

It would be nice to be able to specify which parts of book navigation appear on book pages, such as:
* Table of contents
* Previous/Next/Up links
* Add child page link

## Proposed solution

Add global settings, as follows:

Screen Shot 2022-01-02 at 4 44 20 PM

## Draft of feature description for Press Release (1 paragraph at most)

Backdrop now includes global settings that allow you to select which parts of the book navigation should display on pages included in Books. 

---

**Translation update:**
New strings were introduced: `Display default book navigation on book pages`, `Select book navigation options`, `Table of contents`, `Previous, Next, and Up links`, `Select book links options`, `Add child page`,

### PR DESCRIPTION
**Pull request by herbdool**
Fixes https://github.com/backdrop/backdrop-issues/issues/5430

### CODE REVIEW

File: core/modules/book/book.module [hunk 1]
```diff
@@ -87,11 +87,13 @@ function book_autoload_info() {
...
   $links = array();
+  $config = config('book.settings');
+  $link_options = $config->get('book_links');
 
   if (isset($node->book['depth'])) {
     if ($view_mode == 'full' && node_is_page($node)) {
-      $child_type = config_get('book.settings', 'book_child_type');
-      if ((user_access('add content to books') || user_access('administer book outlines')) && node_access('create', $child_type) && $node->status == 1 && $node->book['depth'] < MENU_MAX_DEPTH) {
+      $child_type = $config->get('book_child_type');
+      if ((user_access('add content to books') || user_access('administer book outlines')) && node_access('create', $child_type) && $node->status == 1 && $node->book['depth'] < MENU_MAX_DEPTH && in_array('book_add_child', $link_options, TRUE)) {
```
Comment (jenlampton) [inline core/modules/book/book.module]: The logic in this if statement is getting hard to follow. Instead of adding more conditions to an already long line, can we rewrite this as something like...

```php
if (user_access('add content to books') || user_access('administer book outlines')) {
  $link_options = $config->get('book_links');
  $create_access = node_access('create', $child_type);
  $depth_allowed =  ($node->book['depth'] < MENU_MAX_DEPTH);
  $show_add_child_link = in_array('book_add_child', $link_options, TRUE);
  if ( $create_access && $node->status == 1 &&  $depth_allowed && $show_add_child_link) {
    ...
  }
}
```

File: core/modules/book/book.theme.inc [hunk 2]
```diff
@@ -36,6 +36,7 @@ function template_preprocess_book_all_books_block(&$variables) {
  * @see book-navigation.tpl.php
  */
 function template_preprocess_book_navigation(&$variables) {
+  $navigation_options = config_get('book.settings','book_navigation_options');
```
Comment (jenlampton) [inline core/modules/book/book.theme.inc]: Can we move this `config_get` inside the check for `if ($book_link['mlid']) {` so we aren't loading it when it's not needed?

**Comment by jenlampton** [comment]
This looks great :)

**Comment by quicksketch** [comment]
Merged into 1.x for 1.21.0. Thanks @herbdool and @jenlampton!
